lib/valkey.rb

```diff
@@ -182,15 +182,32 @@ def convert_response(res, &block)
   end
 
   def send_command(command_type, command_args = [], &block)
+    # Validate connection
+    if @connection.nil?
+      raise "Connection is nil"
+    elsif @connection.null?
+      raise "Connection pointer is null"
+    elsif @connection.address.zero?
+      raise "Connection address is 0"
+    end
+
     channel = 0
     route = ""
 
     route_buf = FFI::MemoryPointer.from_string(route)
 
-    arg_ptrs, arg_lens = build_command_args(command_args)
+    # Handle empty command_args case
+    if command_args.empty?
+      arg_ptrs = FFI::MemoryPointer.new(:pointer, 1)
+      arg_lens = FFI::MemoryPointer.new(:ulong, 1)
+      arg_ptrs.put_pointer(0, FFI::MemoryPointer.new(1))
+      arg_lens.put_ulong(0, 0)
```

Comment:
What this section fixes? the `build_command_args` has to make proper arg_ptrs and arg_lens already